openhexa/cli/api.py

```diff
@@ -191,9 +192,9 @@ def ensure_is_pipeline_dir(pipeline_path: str):
     return True
 
 
-def upload_pipeline(config, pipeline_directory_path: str):
-    pipeline = import_pipeline(pipeline_directory_path)
+def upload_pipeline(config, pipeline_directory_path: str, strategy: ImportStrategy):
     directory = Path(os.path.abspath(pipeline_directory_path))
+    pipeline = import_pipeline(pipeline_directory_path, strategy)
```

Comment:
This is not ideal :

- You call `import_pipeline` with `strategy`,  but if strategy is `AST` there won't be any actual import (unclear)
- `import_pipeline` is a function that returns two very different types of values depending on `strategy`, that you will need to check afterwards.

Instead, we should call `get_pipeline_parameters_specs` with the chosen strategy, and `get_pipeline_parameters_specs` should return the exact same specs with both strategies.

That way, you encapsulate the actual "spec discovery" logic in `get_pipeline_parameters_specs`, and the calling function doesn't need to know what's actually happening in there - it can assume it will receive proper specs.